### PR TITLE
add loading status

### PR DESCRIPTION
**Pull request by shayanbinary**

### CODE REVIEW

File: src/store/LoaderStore.js [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
+
 export default class LoaderStore {
     @observable isActive = false;
+    @observable currentState = false;
+    states = {};
+
+    constructor() {
+        this.states = {
+            'chart-engine': t.translate('Retrieving Chart Engine...'),
+            'market-symbol': t.translate('Retrieving market symbols...'),
```
Comment (MahboobehMohammadi) [inline src/store/LoaderStore.js]: @shayanbinary Can you use all word in caps here? `Retrieving Market Symbols...`

File: src/store/LoaderStore.js [hunk 2]
```diff
@@ -1,7 +1,27 @@
...
+            'trading-time': t.translate('Retrieving trading times...'),
```
Comment (MahboobehMohammadi) [inline src/store/LoaderStore.js]: @shayanbinary Can you use all word in caps here? `Retrieving Trading Times...`

File: sass/components/_loader.scss [hunk 3]
```diff
@@ -1,23 +1,41 @@
...
+        padding-top: 5px;
+        position: absolute;
+        height: 20px;
+        width: 200px;
+        left: 50%;
+        top: 50%;
+        margin-left: -100px;
+        margin-top: 10px;
+        text-align: center;
+    }
```
Comment (MahboobehMohammadi) [inline sass/components/_loader.scss]: @shayanbinary Could you also pls appy the dark theme for both text and loding icon

File: src/store/LoaderStore.js [hunk 4]
```diff
@@ -1,7 +1,27 @@
...
+            'chart-data': t.translate('Retrieving Chart Data...'),
+        };
+    }
+
+    @action.bound setState(_state) {
```
Comment (MahboobehMohammadi) [inline src/store/LoaderStore.js]: @shayanbinary nitpick, we dont use undeline in our naming convention.

File: src/store/ChartStore.js [hunk 5]
```diff
@@ -299,6 +300,7 @@ class ChartStore {
                 }
 
                 this.context = context;
+                this.loader.setState('market-symbol');
```
Comment (MahboobehMohammadi) [inline src/store/ChartStore.js]: @shayanbinary I dont think its the right place to change symbol, try to change the symbol form your test link and you'll see the `Retrieving Chart Data...`

File: src/store/ChartStore.js [hunk 6]
```diff
@@ -339,6 +341,7 @@ class ChartStore {
     }
 
     onMarketOpenClosedChange = (changes) => {
+        this.loader.setState('trading-time');
```
Comment (MahboobehMohammadi) [inline src/store/ChartStore.js]: @shayanbinary And also about `trading-time`, why you put it here? shouldnt it be in `_updateTradeTimes`.

File: src/store/ChartStore.js (hunk 5 above)
Comment (shayanbinary) [reply src/store/ChartStore.js]: @MahboobehMohammadi whenever we change the symbol, new data for active symbol should be fetch from server, so I think that would be typical

File: src/store/ChartStore.js [hunk 7]
```diff
@@ -219,6 +219,7 @@ class ChartStore {
         // TODO: excluded studies
 
         this.loader.show();
+        this.loader.setState('chart-engine');
```
Comment (brucebinary) [inline src/store/ChartStore.js]: you are placing the status messages (all of them) in the wrong place. 

Where you are placing the chart engine loading message - the chart engine has already been loaded.

File: translation/messages.pot [hunk 8]
```diff
@@ -221,6 +221,18 @@ msgstr ""
...
+msgid "Retrieving Chart Data..."
+msgstr ""
+
+msgid "Retrieving Chart Engine..."
+msgstr ""
+
+msgid "Retrieving market symbols..."
+msgstr ""
+
+msgid "Retrieving trading times..."
```
Comment (brucebinary) [inline translation/messages.pot]: this is not up to date

**Pull request merged**